src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfoOfT.cs

```diff
@@ -43,78 +43,89 @@ internal sealed class JsonPropertyInfo<T> : JsonPropertyInfo
             bool isVirtual,
             JsonConverter converter,
             JsonIgnoreCondition? ignoreCondition,
-            JsonNumberHandling? parentTypeNumberHandling,
-            JsonSerializerOptions options)
+            JsonSerializerOptions options,
+            JsonTypeInfo? jsonTypeInfo = null)
         {
-            base.Initialize(
-                parentClassType,
-                declaredPropertyType,
-                converterStrategy,
-                memberInfo,
-                isVirtual,
-                converter,
-                ignoreCondition,
-                parentTypeNumberHandling,
-                options);
-
-            switch (memberInfo)
+            Debug.Assert(converter != null);
+
+            PropertyType = declaredPropertyType;
+            ConverterStrategy = converterStrategy;
+            if (jsonTypeInfo != null)
             {
-                case PropertyInfo propertyInfo:
-                    {
-                        bool useNonPublicAccessors = GetAttribute<JsonIncludeAttribute>(propertyInfo) != null;
+                JsonTypeInfo = jsonTypeInfo;
+            }
 
-                        MethodInfo? getMethod = propertyInfo.GetMethod;
-                        if (getMethod != null && (getMethod.IsPublic || useNonPublicAccessors))
-                        {
-                            HasGetter = true;
-                            Get = options.MemberAccessorStrategy.CreatePropertyGetter<T>(propertyInfo);
-                        }
+            ConverterBase = converter;
+            Options = options;
+            DeclaringType = parentClassType;
+            MemberInfo = memberInfo;
+            IsVirtual = isVirtual;
 
-                        MethodInfo? setMethod = propertyInfo.SetMethod;
-                        if (setMethod != null && (setMethod.IsPublic || useNonPublicAccessors))
+            if (memberInfo != null)
```

Comment:
You can actually embed the null check in the switch statement by doing something like
```C#
switch (memberInfo)
{
     case null:
         break;
     case PropertyInfo propertyInfo:
         ...
```
which should save you a tab of indentation :-)

Reply:
but we also need to do:

```csharp
                // TODO (perf): can we pre-compute some of these values during source gen?
                _converterIsExternalAndPolymorphic = !converter.IsInternalConverter && PropertyType != converter.TypeToConvert;
                PropertyTypeCanBeNull = PropertyType.CanBeNull();
                _propertyTypeEqualsTypeToConvert = typeof(T) == PropertyType;


            GetPolicies(ignoreCondition);
                GetPolicies(ignoreCondition);
```

for `memberInfo != null` case, I'll leave this as is for now